network/state/lookup.go

```diff
@@ -0,0 +1,189 @@
+package state
+
+import (
+	"errors"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/safing/portmaster/network/packet"
+	"github.com/safing/portmaster/network/socket"
+)
+
+// - TCP
+//   - Outbound: Match listeners (in!), then connections (out!)
+//   - Inbound: Match listeners (in!), then connections (out!)
+//   - Clean via connections
+// - UDP
+//   - Any connection: match specific local address or zero IP
+//   - In or out: save direction of first packet:
+//     - map[<local udp bind ip+port>]map[<remote ip+port>]{direction, lastSeen}
+//       - only clean if <local udp bind ip+port> is removed by OS
+//       - limit <remote ip+port> to 256 entries?
+//       - clean <remote ip+port> after 72hrs?
+//       - switch direction to outbound if outbound packet is seen?
+// - IP: Unidentified Process
+
+const (
+	UnidentifiedProcessID = -1
```

Comment:
I reduced the occurences and limited definitions to the `process` and `network/socket` packages. It foremost belongs to the `process` package, but I cannot use that everywhere because of import loops.

Reply:
If we have more constants like that we may think about creating a shared `constants` package